frontend/src/store/index.js

```diff
@@ -161,26 +161,42 @@ const getters = {
       return sortBy(filteredCloudProfiles, 'metadata.name')
     }
   },
-  machineTypesByCloudProfileNameAndZones (state, getters) {
-    return ({ cloudProfileName, zones }) => {
+  machineTypesByCloudProfileName (state, getters) {
+    return ({ cloudProfileName }) => {
+      return getters.machineTypesByCloudProfileNameAndRegionAndZones({ cloudProfileName })
+    }
+  },
+  machineTypesByCloudProfileNameAndRegionAndZones (state, getters) {
+    return ({ cloudProfileName, region, zones }) => {
       const cloudProfile = getters.cloudProfileByName(cloudProfileName)
       if (cloudProfile) {
         const machineTypes = cloudProfile.data.machineTypes
-        const unavailableItems = flatMap(cloudProfile.data.regions, ({ zones }) => {
-          return map(zones, 'unavailableMachineTypes')
+        const regionObject = find(cloudProfile.data.regions, { name: region })
+        const unavailableItems = flatMap(get(regionObject, 'zones', []), zone => {
```

Comment:
extract zones to `zonesForRegion` variable
filter, then flatmap

same below (volumeTypesByCloudProfileName)

Reply:
refactored code for more reuse